lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -1331,8 +1358,743 @@
+                ["day",             twoDigitNumeric],
+                ["hour",            twoDigitNumeric],
+                ["minute",          twoDigitNumeric],
+                ["second",          twoDigitNumeric],
+                ["timeZoneName",    narrowShortLong.slice(1)] // timeZoneName only allows "short" and "long"
+            ];
+
+            /**
+             * From https://unicode.org/repos/cldr/trunk/common/bcp47
+             * TODO(jahorto): we need to load these from the ICU data file in case we don't have them all available
+             */
+            const localeData = {
+                ca(locale) {
+                    const defaultCa = "gregory";
+                    const availableCas = [
+                        "buddhist",
+                        "chinese",
+                        "coptic",
+                        "dangi",
+                        "ethioaa",
+                        "ethiopic-amete-alem", // same as ethioaa
+                        "ethiopic",
+                        "gregory",
+                        "gregorian", // same as gregory
+                        "hebrew",
+                        "indian",
+                        "islamic",
+                        "islamic-umalqura",
+                        "islamic-tbla",
+                        "islamic-civil",
+                        "islamic-rgsa",
+                        "iso8601",
+                        "japanese",
+                        "persian",
+                        "roc",
+                        "islamicc", // same as islamic-civil
+                    ];
+
+                    return _.concat([defaultCa], availableCas);
+                },
+                // ICU doesn't seem to allow us to view all numbering systems available in the data file for a locale
+                // the spec requires at least the following to be supported, so assume that they are?
+                nu(locale) {
+                    const defaultNu = "latn";
+                    const specNus = [
+                        "arab",
+                        "arabext",
+                        "bali",
+                        "beng",
+                        "deva",
+                        "fullwide",
+                        "gujr",
+                        "hanidec",
+                        "khmr",
+                        "knda",
+                        "laoo",
+                        "latn",
+                        "limb",
+                        "mlym",
+                        "mong",
+                        "mymr",
+                        "orya",
+                        "tamldec",
+                        "telu",
+                        "thai",
+                        "tibt",
+                    ];
+                    return _.concat([defaultNu], specNus);
+                },
+                // 12.3.3 - "hc must be << null, h11, h12, h23, h24 >> for all locale values"
```

Comment:
Nit: maybe `#sec-intl.datetimeformat-internal-slots` instead of `12.3.3` based on Doug's comment above